RecoTracker/MkFitCore/src/PropagationMPlex.cc

```diff
@@ -636,19 +642,26 @@ namespace mkfit {
       MPlexQF hitsRl;
       MPlexQF hitsXi;
       MPlexQF propSign;
+
+      // const TrackerInfo &tinfo = *pflags.tracker_info;
+      // Use TrackerInfo for material access ...
 #pragma omp simd
       for (int n = 0; n < NN; ++n) {
         if (n >= N_proc || (noMatEffPtr && noMatEffPtr->constAt(n, 0, 0))) {
           hitsRl(n, 0, 0) = 0.f;
           hitsXi(n, 0, 0) = 0.f;
         } else {
-          const int zbin = Config::materialEff.getZbin(msZ(n, 0, 0));
-          const int rbin = Config::materialEff.getRbin(std::hypot(outPar(n, 0, 0), outPar(n, 1, 0)));
-          hitsRl(n, 0, 0) = (zbin >= 0 && zbin < Config::nBinsZME && rbin >= 0 && rbin < Config::nBinsRME)
-                                ? Config::materialEff.getRlVal(zbin, rbin)
+          const float hypo = std::hypot(outPar(n, 0, 0), outPar(n, 1, 0));
+          const int zbin = (std::abs(msZ(n, 0, 0)) * Config::nBinsZMat) / Config::rangeZMat;
+          const int rbin =
+              (hypo <= Config::rangeRMat)
+                  ? (std::hypot(outPar(n, 0, 0), outPar(n, 1, 0)) * Config::nBinsRMat) / (Config::rangeRMat)
+                  : -1;
+          hitsRl(n, 0, 0) = (zbin >= 0 && zbin < Config::nBinsZMat && rbin >= 0 && rbin < Config::nBinsRMat)
```

Comment:
```suggestion
          bool binsOK = zbin >= 0 && zbin < Config::nBinsZMat && rbin >= 0 && rbin < Config::nBinsRMat;
          hitsRl(n, 0, 0) = binsOK 
```
may be even a bit more readable